cmd/update-main.go

```diff
@@ -17,274 +17,478 @@
+func mcVersionToReleaseTime(version string) (releaseTime time.Time, err *probe.Error) {
+	var e error
+	releaseTime, e = time.Parse(time.RFC3339, version)
+	return releaseTime, probe.NewError(e)
 }
 
-const releaseTagTimeLayout = "2006-01-02T15-04-05Z"
-const mcReleaseURL = "https://dl.minio.io/client/mc/release/" + runtime.GOOS + "-" + runtime.GOARCH + "/"
+// releaseTimeToReleaseTag - converts a time to a string formatted as
+// an official mc release tag.
+//
+// An official mc release tag looks like:
+// `RELEASE.2017-09-29T19-16-56Z`
+func releaseTimeToReleaseTag(releaseTime time.Time) string {
+	return "RELEASE." + releaseTime.Format(mcReleaseTagTimeLayout)
+}
 
-func getCurrentReleaseTime(mcVersion, mcBinaryPath string) (releaseTime time.Time, err *probe.Error) {
-	var e error
-	if releaseTime, e = time.Parse(time.RFC3339, mcVersion); e == nil {
-		return releaseTime, probe.NewError(e)
+// releaseTagToReleaseTime - reverse of `releaseTimeToReleaseTag()`
+func releaseTagToReleaseTime(releaseTag string) (releaseTime time.Time, err *probe.Error) {
+	tagTimePart := strings.TrimPrefix(releaseTag, "RELEASE.")
+	if tagTimePart == releaseTag {
+		return releaseTime, probe.NewError(fmt.Errorf("%s is not a valid release tag", releaseTag))
 	}
+	var e error
+	releaseTime, e = time.Parse(mcReleaseTagTimeLayout, tagTimePart)
+	return releaseTime, probe.NewError(e)
+}
 
-	mcBinaryPath, e = exec.LookPath(mcBinaryPath)
+// getModTime - get the file modification time of `path`
+func getModTime(path string) (t time.Time, err *probe.Error) {
+	// Convert to absolute path
+	absPath, e := filepath.Abs(path)
 	if e != nil {
-		return releaseTime, probe.NewError(e)
+		return t, probe.NewError(fmt.Errorf("Unable to get absolute path of %s. %s", path, e))
 	}
 
-	fi, e := os.Stat(mcBinaryPath)
+	// Version is mc non-standard, we will use mc binary's
+	// ModTime as release time.
+	fi, e := os.Stat(absPath)
 	if e != nil {
-		e = fmt.Errorf("Unable to get ModTime of %s. %s", mcBinaryPath, e)
-	} else {
-		releaseTime = fi.ModTime().UTC()
+		return t, probe.NewError(fmt.Errorf("Unable to get ModTime of %s. %s", absPath, e))
 	}
 
-	return releaseTime, probe.NewError(e)
+	// Return the ModTime
+	return fi.ModTime().UTC(), nil
 }
 
-// GetCurrentReleaseTime - returns this process's release time.  If it is official minio version,
-// parsed version is returned else minio binary's mod time is returned.
+// GetCurrentReleaseTime - returns this process's release time.  If it
+// is official mc version, parsed version is returned else mc
+// binary's mod time is returned.
 func GetCurrentReleaseTime() (releaseTime time.Time, err *probe.Error) {
-	return getCurrentReleaseTime(Version, os.Args[0])
+	if releaseTime, err = mcVersionToReleaseTime(Version); err == nil {
+		return releaseTime, nil
+	}
+
+	// Looks like version is mc non-standard, we use mc
+	// binary's ModTime as release time:
+	return getModTime(os.Args[0])
```

Comment:
no, we shouldn't do that @vadmeste we decided that we will not use 1970 because accidentally things can get replaced.   The current implementation is fine as it is. will fix the path construction.